deploy/lib/server_config.rb

```diff
@@ -2451,7 +2451,8 @@ def xcc
           :http_connection_retry_count => @properties["ml.http.retry-count"].to_i,
           :http_connection_open_timeout => @properties["ml.http.open-timeout"].to_i,
           :http_connection_read_timeout => @properties["ml.http.read-timeout"].to_i,
-          :http_connection_retry_delay => @properties["ml.http.retry-delay"].to_i
+          :http_connection_retry_delay => @properties["ml.http.retry-delay"].to_i,
+          :use_https_for_xcc => @properties["ml.use-https-for-xcc"] == "true"
```

Comment:
So the similar `use-https-for-rest` option checks if either `use-https-for-rest` is `true` _OR_ `ssl-certificate-template` is defined. In that case it makes sense because as soon as you define a certificate template it turns the rest server to `https` when you bootstrap, you don't have to also enable `use-https-for-rest`. For XCC the xml for the definition is built a bit differently, and defining a certificate-template won't also enable ssl on the xcc port. If I were to have the same _OR_ condition I think it would break backwards compatibility. Let me know what you think.

Reply:
Yes, that was exactly the concern I had before. Looks perfect to me..